client/src/components/Chat.jsx

```diff
@@ -52,6 +53,9 @@ const Chat = () => {
 
     const inputRef = useRef('');
 
+    const [lastMessageTime, setLastMessageTime] = useState(null)
+    const [delay, setDelay] = useState(1000);
```

Comment:
ok,
- what we do is we set to keep the time of the last message sent and then we check every second if it has reached the inactiveTimeThreshold or not. We do this instead of checking every 3 minutes (inactiveTimeThreshold) because by doing it that way we would start a new interval every time a message comes for 3 minutes. (multiple setIntervals instances instead of one)
- another approach that I thought was using setTimeOut for 3 minutes and checking at the end if the message came or not. This again would keep on creating new instances of setTimeOut which would be messy instead of checking every second.

Reply:
what i understand is, when we get a new message we start counting if it has reached the threshold or not?

the second option sounds better to me actually, we still run one but whenever a new message comes in it resets, how about that?